### PR TITLE
[SPARK-27222][SQL] Support Instant and LocalDate in Literal.apply

### PR DESCRIPTION
**Pull request by MaxGekk**
## What changes were proposed in this pull request?

In the PR, I propose to extend `Literal.apply` to support constructing literals of `TimestampType` and `DateType` from `java.time.Instant` and `java.time.LocalDate`. The java classes have been already supported as external types for `TimestampType` and `DateType` by the PRs #23811  and #23913. 

## How was this patch tested?

Added new tests to `LiteralExpressionSuite`.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
ok to test

**Comment by HyukjinKwon** [comment]
add to whitelist

**Comment by MaxGekk** [comment]
@cloud-fan Could you take a look at this PR, please.

**Review by HyukjinKwon — APPROVED** [review]

**Review by srowen — COMMENTED** [review]
Seems reasonable to me, but I don't know this part well

**Comment by cloud-fan** [comment]
thanks, merging to master!

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/LiteralExpressionSuite.scala [hunk 1]
```diff
@@ -228,4 +236,47 @@ class LiteralExpressionSuite extends SparkFunSuite with ExpressionEvalHelper {
...
+      LocalDate.of(1969, 12, 31),
+      LocalDate.of(1970, 1, 1),
+      LocalDate.of(2019, 3, 20),
+      LocalDate.of(2100, 5, 17)).foreach { localDate =>
+      checkEvaluation(Literal(localDate), localDate)
+    }
+  }
+
+  test("construct literals from arrays of java.time.LocalDate") {
+    withSQLConf(SQLConf.DATETIME_JAVA8API_EANBLED.key -> "true") {
```
Comment (cloud-fan) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/LiteralExpressionSuite.scala]: not a big idea, but the behavior of literal is not controlled by this config right?

**Pull request closed without merging**